crates/duckdb/examples/hello-ext-capi/main.rs

```diff
@@ -0,0 +1,93 @@
+extern crate duckdb;
+extern crate duckdb_loadable_macros;
+extern crate libduckdb_sys;
+
+use duckdb::{
+    core::{DataChunkHandle, Inserter, LogicalTypeHandle, LogicalTypeId},
+    vtab::{BindInfo, Free, FunctionInfo, InitInfo, VTab},
+    Connection, Result,
+};
+use duckdb_loadable_macros::duckdb_entrypoint_c_api;
+use libduckdb_sys as ffi;
+use std::{
+    error::Error,
+    ffi::{c_char, c_void, CString},
+};
+
+#[repr(C)]
+struct HelloBindData {
+    name: *mut c_char,
+}
+
+impl Free for HelloBindData {
+    fn free(&mut self) {
+        unsafe {
+            if self.name.is_null() {
```

Comment:
the default value of this pointer is not null, its some undefined value because the memory is allocated in C and passed across the ffi boundary. this branch will probably never be hit. seems like this free trait is implemented everywhere incorrectly in this project though so 🤷  just a heads up

Reply:
also your code doesn't ever not initialize name

Reply:
I just blindly copied `examples/hello-ext` I will leave it to a future pr to clean this up!